### PR TITLE
VZ-10840 change internal module versions to match Verrazzano version

### PR DESCRIPTION
**Pull request by mbenmclean**
This change changes the internal modules versions in the catalog to match the Verrazzano version. The catalog is generated  at build time similar to the how the bom is generated. This change also moved the catalog into the manifests directory.

### CODE REVIEW

File: platform-operator/manifests/catalog/catalog.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ modules:
   - name: cert-manager-webhook-oci
     version: 0.1.0
   - name: cluster-issuer
-    version: 2.0.0
+    version: VERRAZZANO_VERSION
```
Comment (pfmackin) [inline platform-operator/manifests/catalog/catalog.yaml:12]: Where are you evaluating this?

File: tools/scripts/generate_catalog.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
```
Comment (michael-cico) [inline tools/scripts/generate_catalog.sh:1]: Could we combine this with the generate_bom.sh script?  i.e. consolidate the processing into one place?

Don't have to for this pass.

File: platform-operator/manifests/catalog/catalog.yaml (hunk 1 above)
Comment (michael-cico) [reply platform-operator/manifests/catalog/catalog.yaml:12]: or provide a path-to-catalog argument to the operator like we do with the BOM.  Then you can point that to a locally generated catalog also.

File: platform-operator/Makefile [hunk 3]
```diff
@@ -225,3 +225,8 @@ generate-bom:
 	mkdir out || true
 	../tools/scripts/generate_bom.sh verrazzano-bom.json ${VERRAZZANO_VERSION} ${VERRAZZANO_APPLICATION_OPERATOR_IMAGE} ${VERRAZZANO_CLUSTER_OPERATOR_IMAGE} ${VERRAZZANO_AUTHPROXY_IMAGE} ${DOCKER_IMAGE_NAME} ${DOCKER_IMAGE_TAG} out/generated-verrazzano-bom.json
 
+.PHONY: generate-catalog
```
Comment (michael-cico) [inline platform-operator/Makefile:228]: if we can consolidate this with the BOM script/target and have one that would be good.  Can be a follow-up

File: platform-operator/manifests/catalog/catalog.yaml (hunk 1 above)
Comment (mbenmclean) [reply platform-operator/manifests/catalog/catalog.yaml:12]: Hmm, well the VPO code don't know what the version is. I could some code to check if the version is VERRAZZANO_VERSION and if so use the version from the BOM. That will allow you to set the VZ_INSTALL_VERSION in the debugger.

File: tools/scripts/generate_catalog.sh (hunk 2 above)
Comment (mbenmclean) [reply tools/scripts/generate_catalog.sh:1]: I thought about them having them together but decided I preferred separate. It allows for more flexibility.

**Review by michael-cico — APPROVED** [review]

**Pull request merged**